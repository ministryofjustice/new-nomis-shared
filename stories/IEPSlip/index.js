/* eslint-disable import/no-extraneous-dependencies */
import React from 'react'
import { storiesOf } from '@storybook/react'
import IEPSlip from '../../components/IEPSlip'

const props = {
  raisedBy: 'An Officer',
  issuedBy: 'Another Officer',
  offenderName: 'An Offender',
  offenderNo: 'ABC123',
  caseNote: 'Refused: Did not want to work',
  cellLocation: 'LEI-123',
}

storiesOf('IEP slip', module)
  .add('Component default', () => <IEPSlip {...props} />)
  .add('With amendments', () => (
    <IEPSlip
      {...props}
      amendments={[
        { authorName: 'Amendment Author', creationDateTime: '2019-01-01T14:17:30', additionalNoteText: 'Rescinded' },
        { authorName: 'Amendment Author', creationDateTime: '2019-01-01T15:17:30', additionalNoteText: 'Updated' },
      ]}
    />
  ))
