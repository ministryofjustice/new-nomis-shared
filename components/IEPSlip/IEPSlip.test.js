import React from 'react'
import renderer from 'react-test-renderer'

import IEPSlip from '.'

const props = {
  raisedBy: 'An Officer',
  offenderName: 'An Offender',
  offenderNo: 'ABC123',
  caseNote: 'Refused: Did not want to work',
  cellLocation: 'LEI-123',
}

describe('<IEPSlip />', () => {
  Date.now = jest.fn(() => new Date(Date.UTC(2019, 0, 1)).valueOf())
  it('renders correctly', () => {
    const tree = renderer.create(<IEPSlip {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with an activity or area', () => {
    const tree = renderer.create(<IEPSlip {...props} activityName="Gym" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with an issued by name', () => {
    const tree = renderer.create(<IEPSlip {...props} activityName="Gym" issuedBy="Another Officer" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with amendments', () => {
    const tree = renderer
      .create(
        <IEPSlip
          {...props}
          amendments={[
            {
              authorName: 'Amendment Author',
              creationDateTime: '2019-01-01T14:17:30',
              additionalNoteText: 'Rescinded',
            },
            { authorName: 'Amendment Author', creationDateTime: '2019-01-01T15:17:30', additionalNoteText: 'Updated' },
          ]}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
