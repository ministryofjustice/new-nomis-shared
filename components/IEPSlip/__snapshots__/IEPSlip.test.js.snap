// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IEPSlip /> renders correctly 1`] = `
<div
  className="sc-bwzfXH cQUzRp"
>
  <div
    className="sc-htpNat jyGGsS"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      <h2
        className="sc-bdVaJa YBcoz"
        size="MEDIUM"
      >
        IEP Warning
      </h2>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Date and time: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        1 January 2019, 12:00 am
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat jyGGsS"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Name: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        An Offender
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Number: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        ABC123
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Location: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        LEI-123
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat jyGGsS"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Reason:
      <span
        className="sc-ifAKCX hxQqKr"
      >
        Refused: Did not want to work
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat garCfj"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Raised by: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        A Officer
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Date: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        1 January 2019
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat garCfj"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Issued by: 
      <span
        className="sc-ifAKCX fJmoaE"
      />
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Date: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        1 January 2019
      </span>
    </div>
  </div>
</div>
`;

exports[`<IEPSlip /> renders with amendments 1`] = `
<div
  className="sc-bwzfXH cQUzRp"
>
  <div
    className="sc-htpNat jyGGsS"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      <h2
        className="sc-bdVaJa YBcoz"
        size="MEDIUM"
      >
        IEP Warning
      </h2>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Date and time: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        1 January 2019, 12:00 am
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat jyGGsS"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Name: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        An Offender
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Number: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        ABC123
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Location: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        LEI-123
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat jyGGsS"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Reason:
      <span
        className="sc-ifAKCX hxQqKr"
      >
        Refused: Did not want to work
      </span>
      <div
        className="sc-EHOje bVSyRj"
      >
        Amendment: 
        <span
          className="sc-ifAKCX hxQqKr"
        >
          Rescinded
        </span>
        <div>
          1 January 2019, 2:17 pm
        </div>
        <div>
          A Author
        </div>
      </div>
      <div
        className="sc-EHOje bVSyRj"
      >
        Amendment: 
        <span
          className="sc-ifAKCX hxQqKr"
        >
          Updated
        </span>
        <div>
          1 January 2019, 3:17 pm
        </div>
        <div>
          A Author
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-htpNat garCfj"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Raised by: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        A Officer
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Date: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        1 January 2019
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat garCfj"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Issued by: 
      <span
        className="sc-ifAKCX fJmoaE"
      />
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Date: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        1 January 2019
      </span>
    </div>
  </div>
</div>
`;

exports[`<IEPSlip /> renders with an activity or area 1`] = `
<div
  className="sc-bwzfXH cQUzRp"
>
  <div
    className="sc-htpNat jyGGsS"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      <h2
        className="sc-bdVaJa YBcoz"
        size="MEDIUM"
      >
        IEP Warning
      </h2>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Date and time: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        1 January 2019, 12:00 am
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat jyGGsS"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Name: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        An Offender
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Number: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        ABC123
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Location: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        LEI-123
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Activity or area: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        Gym
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat jyGGsS"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Reason:
      <span
        className="sc-ifAKCX hxQqKr"
      >
        Refused: Did not want to work
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat garCfj"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Raised by: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        A Officer
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Date: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        1 January 2019
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat garCfj"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Issued by: 
      <span
        className="sc-ifAKCX fJmoaE"
      />
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Date: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        1 January 2019
      </span>
    </div>
  </div>
</div>
`;

exports[`<IEPSlip /> renders with an issued by name 1`] = `
<div
  className="sc-bwzfXH cQUzRp"
>
  <div
    className="sc-htpNat jyGGsS"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      <h2
        className="sc-bdVaJa YBcoz"
        size="MEDIUM"
      >
        IEP Warning
      </h2>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Date and time: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        1 January 2019, 12:00 am
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat jyGGsS"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Name: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        An Offender
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Number: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        ABC123
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Location: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        LEI-123
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Activity or area: 
      <span
        className="sc-ifAKCX hxQqKr"
      >
        Gym
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat jyGGsS"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Reason:
      <span
        className="sc-ifAKCX hxQqKr"
      >
        Refused: Did not want to work
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat garCfj"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Raised by: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        A Officer
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Date: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        1 January 2019
      </span>
    </div>
  </div>
  <div
    className="sc-htpNat garCfj"
  >
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Issued by: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        A Officer
      </span>
    </div>
    <div
      className="sc-bxivhb hEVLlJ"
    >
      Date: 
      <span
        className="sc-ifAKCX fJmoaE"
      >
        1 January 2019
      </span>
    </div>
  </div>
</div>
`;
